File: manage/src/api/member.ts
import axios from 'axios'
import request from './request'

export interface MemberItem {
  id: number
  username: string
  nickname: string
  email: string
  mobile: string
  status: number
  create_time: string
}

export function fetchMemberList(params: {
  page?: number
  size?: number
  username?: string
  status?: number
}) {
  const q: Record<string, unknown> = { page: params.page ?? 1, size: params.size ?? 10 }
  if (params.username) q.username = params.username
  if (params.status !== undefined && params.status >= 0) q.status = params.status
  return request.get<unknown, { data?: { list?: MemberItem[]; total?: number } }>('/admin/member/list', {
    params: q,
  })
}

export function createMember(payload: {
  username: string
  password: string
  nickname?: string
  email?: string
  mobile?: string
  status?: number
}) {
  return request.post<unknown, { data?: { id?: number } }>('/admin/member', payload)
}

export function updateMember(
  id: number,
  payload: { password?: string; nickname?: string; email?: string; mobile?: string; status?: number }
) {
  return request.put<unknown, unknown>(`/admin/member/${id}`, payload)
}

export function deleteMember(id: number) {
  return request.delete<unknown, unknown>(`/admin/member/${id}`)
}

export interface MemberImportResult {
  total: number
  success: number
  failed: number
  errors: string[]
}

/** multipart 导入客户（表单字段名 file，CSV UTF-8；用户名规则由表单参数传入） */
export function importMembersCsv(
  file: File,
  params: {
    country: string
    year: string
    seq_digits: number
    use_random_password?: boolean
    email_pick_positions?: string
    fixed_password_suffix?: string
    send_password_notice?: boolean
  }
) {
  const fd = new FormData()
  fd.append('country', params.country)
  fd.append('year', params.year)
  fd.append('seq_digits', String(params.seq_digits))
  if (params.use_random_password !== undefined) {
    fd.append('use_random_password', String(params.use_random_password))
  }
  if (params.email_pick_positions) {
    fd.append('email_pick_positions', params.email_pick_positions)
  }
  if (params.fixed_password_suffix) {
    fd.append('fixed_password_suffix', params.fixed_password_suffix)
  }
  if (params.send_password_notice !== undefined) {
    fd.append('send_password_notice', String(params.send_password_notice))
  }
  fd.append('file', file)
  return request.post<unknown, { data?: MemberImportResult }>('/admin/member/import', fd)
}

/** 下载客户导入 CSV 模板（不经统一 JSON 包装） */
export async function downloadMemberImportTemplate() {
  const token = localStorage.getItem('admin_token')
  const res = await axios.get('/api/admin/member/import-template', {
    responseType: 'blob',
    timeout: 60000,
    headers: token ? { Authorization: `Bearer ${token}` } : {},
    validateStatus: () => true,
  })
  if (res.status === 401) {
    throw new Error('未登录或登录已过期')
  }
  const ct = res.headers['content-type'] || ''
  if (ct.includes('application/json')) {
    const text = await (res.data as Blob).text()
    let msg = '下载失败'
    try {
      const j = JSON.parse(text) as { message?: string }
      if (j.message) msg = j.message
    } catch {
      /* */
    }
    throw new Error(msg)
  }
  if (res.status !== 200) {
    throw new Error(`下载失败 (${res.status})`)
  }
  const blob = res.data as Blob
  const url = URL.createObjectURL(blob)
  const a = document.createElement('a')
  a.href = url
  a.download = '客户导入模板.csv'
  a.rel = 'noopener'
  document.body.appendChild(a)
  a.click()
  a.remove()
  URL.revokeObjectURL(url)
}

File: api/admin/member/v1/member.go
package v1

import (
	"github.com/gogf/gf/v2/frame/g"
	"github.com/gogf/gf/v2/net/ghttp"
)

type MemberListReq struct {
	g.Meta   `path:"/member/list" method:"get" tags:"会员" summary:"会员列表" permission:"member:list"`
	Page     int    `json:"page" dc:"页码"`
	Size     int    `json:"size" dc:"每页条数"`
	Username string `json:"username" dc:"用户名"`
	Status   int    `json:"status" dc:"状态：0正常 1停用"`
}

type MemberListRes struct {
	List  []*MemberItem `json:"list" dc:"列表"`
	Total int           `json:"total" dc:"总数"`
}

type MemberItem struct {
	Id         int64  `json:"id" dc:"会员ID"`
	Username   string `json:"username" dc:"用户名"`
	Nickname   string `json:"nickname" dc:"昵称"`
	Email      string `json:"email" dc:"邮箱"`
	Mobile     string `json:"mobile" dc:"手机号"`
	Status     int    `json:"status" dc:"状态：0正常 1停用"`
	CreateTime string `json:"create_time" dc:"创建时间"`
}

type MemberCreateReq struct {
	g.Meta   `path:"/member" method:"post" tags:"会员" summary:"新增会员" permission:"member:create"`
	Username string `json:"username" v:"required#err.invalid_params" dc:"用户名"`
	Password string `json:"password" v:"required#err.invalid_params" dc:"密码"`
	Nickname string `json:"nickname" dc:"昵称"`
	Email    string `json:"email" dc:"邮箱"`
	Mobile   string `json:"mobile" dc:"手机号"`
	Status   int    `json:"status" dc:"状态：0正常 1停用"`
}

type MemberCreateRes struct {
	Id int64 `json:"id" dc:"会员ID"`
}

type MemberUpdateReq struct {
	g.Meta   `path:"/member/{id}" method:"put" tags:"会员" summary:"更新会员" permission:"member:update"`
	Id       int64  `json:"id" in:"path" v:"required#err.invalid_params" dc:"会员ID"`
	Password string `json:"password" dc:"密码"`
	Nickname string `json:"nickname" dc:"昵称"`
	Email    string `json:"email" dc:"邮箱"`
	Mobile   string `json:"mobile" dc:"手机号"`
	Status   int    `json:"status" dc:"状态：0正常 1停用"`
}

type MemberUpdateRes struct{}

type MemberDeleteReq struct {
	g.Meta `path:"/member/{id}" method:"delete" tags:"会员" summary:"删除会员" permission:"member:delete"`
	Id     int64 `json:"id" in:"path" v:"required#err.invalid_params" dc:"会员ID"`
}

type MemberDeleteRes struct{}

type MemberImportReq struct {
	g.Meta              `path:"/member/import" method:"post" tags:"会员" summary:"批量导入客户(CSV)" permission:"member:import"`
	Country             string            `json:"country" form:"country" dc:"自动生成用户名：国家标识，如 TH"`
	Year                string            `json:"year" form:"year" dc:"自动生成用户名：年份，如 2026"`
	SeqDigits           int               `json:"seq_digits" form:"seq_digits" dc:"自动生成用户名：序号位数（≥1）"`
	UseRandomPassword   bool              `json:"use_random_password" form:"use_random_password" dc:"密码为空时是否使用随机密码（false=邮箱规则）"`
	EmailPickPositions  string            `json:"email_pick_positions" form:"email_pick_positions" dc:"固定规则取邮箱位次（1-based），如 1,3,5"`
	FixedPasswordSuffix string            `json:"fixed_password_suffix" form:"fixed_password_suffix" dc:"固定规则后缀，默认 hskmock"`
	SendPasswordNotice  bool              `json:"send_password_notice" form:"send_password_notice" dc:"导入成功后是否发送账号密码通知邮件（模板 forget_password）"`
	File                *ghttp.UploadFile `json:"file" type:"file" dc:"CSV 文件（表单字段名 file）"`
}

type MemberImportRes struct {
	Total   int      `json:"total" dc:"有效数据行数（不含空行）"`
	Success int      `json:"success" dc:"成功条数"`
	Failed  int      `json:"failed" dc:"失败条数"`
	Errors  []string `json:"errors" dc:"失败明细（含行号，最多返回若干条）"`
}
